### PR TITLE
[WIP] Add Java Flight sample application

### PR DESCRIPTION
**Pull request by keeratsingh**
- Added Unit Tests.
- Added ability to take parameters from command line.
- Added Github Workflow for Java testing.

### CODE REVIEW

File: .github/workflows/java-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        # Python is needed for readiness check.
+        python-version: [3.8]
+        java-version: [1.8]
+
+    services:
+      dremio:
+        image: dremio/dremio-oss:latest
+        ports:
+          - 9047:9047
+          - 31010:31010
```
Comment (tifflhl) [inline .github/workflows/java-workflow.yaml:48]: I think we don't need to expose the jdbc port.

File: .github/workflows/python-workflow.yaml [hunk 2]
```diff
@@ -46,6 +46,7 @@ jobs:
         image: dremio/dremio-oss:latest
         ports:
           - 9047:9047
+          - 31010:31010
```
Comment (tifflhl) [inline .github/workflows/python-workflow.yaml:49]: Same as above.

File: java/JavaAdhocFlightClient.iml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tifflhl) [inline java/JavaAdhocFlightClient.iml:1]: Is the iml file needed?

File: java/pom.xml [hunk 4]
```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>adhoc</groupId>
+    <artifactId>java-flight-client</artifactId>
```
Comment (tifflhl) [inline java/pom.xml:8]: java-flight-sample-client-application

**Review by tifflhl — COMMENTED** [review]
I think we should add funcitionality for client properties as well?

File: .github/workflows/java-workflow.yaml (hunk 1 above)
Comment (keeratsingh) [reply .github/workflows/java-workflow.yaml]: Done.

File: .github/workflows/python-workflow.yaml (hunk 2 above)
Comment (keeratsingh) [reply .github/workflows/python-workflow.yaml]: Done.

File: java/pom.xml (hunk 4 above)
Comment (keeratsingh) [reply java/pom.xml]: Removed.

File: java/JavaAdhocFlightClient.iml (hunk 3 above)
Comment (keeratsingh) [reply java/JavaAdhocFlightClient.iml]: Removed.

**Comment by tifflhl** [comment]
This PR can be closed, changes are now tracked in https://github.com/dremio-hub/arrow-flight-client-examples/pull/10

**Pull request closed without merging**